presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/DbResourceGroupConfigurationManager.java

```diff
@@ -0,0 +1,321 @@
+            }
+        }
+
+        synchronized (groups) {
+            // In case there is an exception later in this method, ensure current resource group config is accurate
+            for (String deletedGroup : Sets.difference(this.groupMap.keySet(), newGroupMap.keySet())) {
+                for (ResourceGroupId groupId : activeGroups.getOrDefault(deletedGroup, ImmutableList.of())) {
+                    disableGroup(groups.get(groupId));
+                }
+            }
+            configureChangedGroups(changedGroups);
+            this.groupMap = newGroupMap;
+            ImmutableList.Builder<ResourceGroupRecord> rootGroupBuilder = ImmutableList.builder();
+            for (String root : rootGroupNames) {
+                rootGroupBuilder.add(groupMap.get(root));
+            }
+            this.cpuQuotaPeriod.set(cpuQuotaPeriod);
+            this.rootGroups.set(rootGroupBuilder.build());
+            loadSelectors();
+        }
+    }
+
+    private void loadSelectors()
+    {
+        synchronized (groups) {
+            List<SelectorRecord> selectorRecords = dao.getSelectors();
+            ImmutableList.Builder<ResourceGroupSelector> selectorBuilder = ImmutableList.builder();
+            Set<SelectorRecord> selectorSet = new HashSet<>();
+            for (SelectorRecord selectorRecord : selectorRecords) {
+                checkState(groupMap.containsKey(selectorRecord.getResourceGroup()), format("Invalid resource group specified: %s", selectorRecord.getResourceGroup()));
+                checkState(selectorSet.add(selectorRecord), format("Duplicate selector specified: %s", selectorRecord));
+                selectorBuilder.add(selectorRecord.toStaticSelector());
+            }
+            this.selectors.set(selectorBuilder.build());
+        }
+    }
+
+    private void configureChangedGroups(Map<String, ResourceGroupRecord> changedGroups)
+    {
+        synchronized (groups) {
+            for (Map.Entry<String, ResourceGroupRecord> entry : changedGroups.entrySet()) {
+                ResourceGroupRecord groupRecord = entry.getValue();
+                for (ResourceGroupId groupId : activeGroups.getOrDefault(entry.getKey(), ImmutableList.of())) {
+                    ResourceGroup group = groups.get(groupId);
+                    configureGroup(group, groupRecord);
+                }
+                // Temporarily insert updated resource group record to guarantee current config is accurate
+                // groupMap will be replaced as soon as this function returns
+                groupMap.put(entry.getKey(), entry.getValue());
+            }
+        }
+    }
+
+    private void disableGroup(ResourceGroup group)
+    {
+        synchronized (group) {
+            // Disable groups that are removed from the db
+            group.setMaxRunningQueries(0);
+            group.setMaxQueuedQueries(0);
+        }
+    }
+
+    private void configureGroup(ResourceGroup group, ResourceGroupRecord groupRecord)
```

Comment:
This code seems almost identical to the configuration code in `FileResourceGroupConfigurationManager`. Is it possible to refactor that code into a helper method that can be used in both places?